### DISCUSSION TITLE
Direto XR ( Zwift Cog with Zwift Click ) - Rouvy compatibility

### DISCUSSION THREAD

**Discussion by MiMoravec** [cagnulein/qdomyos-zwift#2599] [General]
Hi, this might be double-post, but could you please confirm, if new Direto XR ( Zwift Cog with Zwift Click ) with QZ app will allow virtual shifting in Rouvy?
Rouvy is running on MacBook, QZ will be installed on iPhone.

**Comment by cagnulein** [comment]
yes it does @MiMoravec !

**Reply by MiMoravec** [reply]
Many thanks for quick reply. Let me extend the compatibility question - does Zwift Cog and Click uses same protocol as Zwift Ride frame (with shifters)? I'm not sure, how far should I go with setup upgrade for winter. 
Rouvy is the only app where I care about the virtual shifting, I'm using just ERG more in other apps.

**Reply by cagnulein** [reply]
i can tell you that from the qz point of view all the trainers are using the same protocol, so qz read the zwift click event and it sends it to the trainer + the gear value. Zwift ride has also a special command for gear that I'm implementing in these days, but at the end of the day, it's always a virtual gear, so same principle
